components/base/BaseTag.vue

```diff
@@ -43,289 +43,75 @@ const props = withDefaults(
-          'border-muted-300 text-muted-600 dark:bg-muted-800 dark:border-muted-700 dark:text-muted-300 border bg-white'
-        )
-        break
-      }
-      case 'muted': {
-        switch (props.flavor) {
-          case 'solid': {
-            classes.push(
-              'text-muted-500 bg-muted-200 dark:bg-muted-700 dark:text-white'
-            )
-          }
-          case 'pastel': {
-            classes.push(
-              'text-muted-500 bg-muted-100 dark:bg-muted-800 dark:text-muted-200'
-            )
-          }
-        }
-        break
-      }
-      case 'primary': {
-        switch (props.flavor) {
-          case 'solid': {
-            classes.push('bg-primary-500 dark:bg-primary-500 text-white')
-
-            break
-          }
-          case 'pastel': {
-            classes.push(
-              'bg-primary-100 text-primary-500 border-primary-100 dark:border-primary-500 dark:text-primary-500 border dark:bg-transparent'
-            )
-
-            break
-          }
-          case 'outline': {
-            classes.push('text-primary-500 border-primary-500 border ')
-
-            break
-          }
-        }
-        break
-      }
-      case 'info': {
-        switch (props.flavor) {
-          case 'solid': {
-            classes.push('bg-info-500 dark:bg-info-500 text-white')
-
-            break
-          }
-          case 'pastel': {
-            classes.push(
-              'bg-info-100 text-info-500 border-info-100 dark:border-info-500 dark:text-info-500 border dark:bg-transparent'
-            )
-
-            break
-          }
-          case 'outline': {
-            classes.push('text-info-500 border-info-500 border')
-
-            break
-          }
-        }
-        break
-      }
-      case 'success': {
-        switch (props.flavor) {
-          case 'solid': {
-            classes.push('bg-success-500 dark:bg-success-500  text-white')
-
-            break
-          }
-          case 'pastel': {
-            classes.push(
-              'bg-success-100 text-success-500 border-success-100 dark:border-success-500 dark:text-success-500 border dark:bg-transparent'
-            )
-
-            break
-          }
-          case 'outline': {
-            classes.push('text-success-500  border-success-500 border')
-
-            break
-          }
-        }
-        break
-      }
-      case 'warning': {
-        switch (props.flavor) {
-          case 'solid': {
-            classes.push('bg-warning-500 dark:bg-warning-500 text-white')
-
-            break
-          }
-          case 'pastel': {
-            classes.push(
-              'bg-warning-100  text-warning-500 border-warning-100 dark:border-warning-500 dark:text-warning-500 border dark:bg-transparent'
-            )
-
-            break
-          }
-          case 'outline': {
-            classes.push('text-warning-500  border-warning-500 border')
+const solidColorStyle = computed(() => [
+  `bg-${props.color}-500`,
```

Comment:
[Don't construct class names dynamically](https://tailwindcss.com/docs/content-configuration#dynamic-class-names) as tailwind JIT engine will not add them to final bundle.

We can do someting similar to:

```ts
const colorVariants = {
  primary: 'bg-primary-600 hover:bg-primary-500',
  warning: 'bg-warning-600 hover:bg-warning-500',
  // ...
}
const solidColorStyle = computed(() => [
  return colorVariants[props.color]
])

```